processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -366,7 +404,7 @@ public int compare(RowBasedKey key1, RowBasedKey key2)
     private static int compareDimsInRows(RowBasedKey key1, RowBasedKey key2, int dimStart)
     {
       for (int i = dimStart; i < key1.getKey().length; i++) {
-        final int cmp = ((String) key1.getKey()[i]).compareTo((String) key2.getKey()[i]);
+        final int cmp = ((Comparable) key1.getKey()[i]).compareTo(key2.getKey()[i]);
```

Comment:
I tried changing the key within RowBasedKey to `Comparable[]` instead of `Object[]` and that results in some json deserialization errors, I think I'll leave it as is for this PR